x-pack/plugins/apm/public/components/app/correlations/use_failed_transactions_correlations.ts

```diff
@@ -0,0 +1,252 @@
+    getReducer<FailedTransactionsCorrelationsResponse & CorrelationsProgress>(),
+    getInitialResponse()
+  );
+  const setResponse = useMemo(
+    () => debounce(setResponseUnDebounced, DEBOUNCE_INTERVAL),
+    []
+  );
+
+  // We're using a ref here because otherwise the startFetch function might have
+  // a stale value for checking if the task has been cancelled.
+  const isCancelledRef = useRef(false);
+
+  const startFetch = useCallback(async () => {
+    abortCtrl.current.abort();
+    abortCtrl.current = new AbortController();
+    isCancelledRef.current = false;
+
+    setResponse({
+      ...getInitialResponse(),
+      isRunning: true,
+      // explicitly set these to undefined to override a possible previous state.
+      error: undefined,
+      failedTransactionsCorrelations: undefined,
+      percentileThresholdValue: undefined,
+      overallHistogram: undefined,
+      errorHistogram: undefined,
+      fieldStats: undefined,
+    });
+    setResponse.flush();
+
+    try {
+      // `responseUpdate` will be enriched with additional data with subsequent
+      // calls to the overall histogram, field candidates, field value pairs, correlation results
+      // and histogram data for statistically significant results.
+      const responseUpdate: FailedTransactionsCorrelationsResponse = {
+        ccsWarning: false,
+      };
+
+      // Initial call to fetch the overall distribution for the log-log plot.
+      const { overallHistogram, percentileThresholdValue } = await callApmApi({
+        endpoint: 'POST /internal/apm/latency/overall_distribution',
+        signal: abortCtrl.current.signal,
+        params: {
+          body: {
+            ...fetchParams,
+            percentileThreshold: DEFAULT_PERCENTILE_THRESHOLD,
+          },
+        },
+      });
+      responseUpdate.overallHistogram = overallHistogram;
+      responseUpdate.percentileThresholdValue = percentileThresholdValue;
+
+      const { overallHistogram: errorHistogram } = await callApmApi({
+        endpoint: 'POST /internal/apm/latency/overall_distribution',
+        signal: abortCtrl.current.signal,
+        params: {
+          body: {
+            ...fetchParams,
+            percentileThreshold: DEFAULT_PERCENTILE_THRESHOLD,
+            termFilters: [
+              { fieldName: EVENT_OUTCOME, fieldValue: EventOutcome.failure },
+            ],
+          },
+        },
+      });
+      responseUpdate.errorHistogram = errorHistogram;
+
+      if (isCancelledRef.current) {
+        return;
+      }
+
+      setResponse({
+        ...responseUpdate,
+        loaded: LOADED_OVERALL_HISTOGRAM,
+      });
+      setResponse.flush();
+
+      const { fieldCandidates: candidates } = await callApmApi({
+        endpoint: 'GET /internal/apm/correlations/field_candidates',
+        signal: abortCtrl.current.signal,
+        params: {
+          query: fetchParams,
+        },
+      });
+
+      if (isCancelledRef.current) {
```

Comment:
why is  this necessary? wouldn't the same result be achieved by reinitializing the abort controller? and what happens if you pass an aborted signal to a fetch call? will it execute or be immediately aborted?

Reply:
`abortController` is used to be passed on to cancel individual requests that are already running. `isCancelledRef` is used to cancel the overall progress of all the tasks that are run within `startFetch()`. I added a comment about it in 72bcc11aaa2 to clarify. Does that explain it? Let me know if I might be missing something about the capabilities of `abortController`, could we use `abortController` itself in this check here?

Reply:
Yes, you can use controller.signal.aborted: https://developer.mozilla.org/en-US/docs/Web/API/AbortSignal. But you can also simply pass in the signal, and the request will immediately be aborted, which will then automatically be handled by your catch clause.

Reply:
Great, that let me get rid of `isCancelledRef` in 97890cf.